brave/brave-ios
# Fix #7585: Portfolio filters and display settings

## Summary of Changes
- Added `Filters and Display Settings` modal with multiple filter options & persistence to Portfolio:
    - Change sort, hide small balances, filter by accounts and by networks.
- Refactored `AccountListView` to support multiple selection, similar style to networks. 
    - There is a shared root view, a filter view and a selection view. 
    - `NavigationView` was pulled out so we can push it as a detail view

This pull request fixes #7585

## Submitter Checklist:


## Test Plan:
NOTE: 60% presentation size is a nice to have only available on iOS 16 and later.

### Test `Sort Assets`:
1. Go to Portfolio and tap the filters button.
2. Change the `Sort Assets` option and tap `Save Changes`.
3. The portfolio should reflect your selected option.
4. Test this for all options.

https://github.com/brave/brave-ios/assets/5314553/b8b5536a-7f0a-4110-849e-e07144820494

### Test `Hide Small Balances`:
1. Go to Portfolio and tap the filters button.
2. Toggle on the `Hide Small Balances` toggle and tap `Save Changes`.
3. Verify the portfolio hide assets with a balance less than $1.

https://github.com/brave/brave-ios/assets/5314553/97887ab7-4357-4de4-a967-1d7a7397b3b8

### Test `Select Accounts`
1. Go to Portfolio and tap the filters button.
2. Tap `Select Accounts`.
3. De-select some accounts and tap `Save Changes`.
4. Verify the balances are updated to reflect the new accounts.
    - Tip: If you de-select all Solana accounts, SOL should show as 0.
    - Tip: If you have multiple accounts for one coin type, you can tap to Asset Detail for an asset to see the balance for ALL accounts which can be used to help verify only the selected accounts balance(s) are shown in Portfolio.

https://github.com/brave/brave-ios/assets/5314553/de802b3a-1fe6-4a9b-ac51-e44bf0059733

### Testing `Select Networks`:
1. Go to Portfolio and tap the filters button.
2. Tap `Select Networks`.
3. De-select some networks and tap `Save Changes`.
4. Verify the assets from the de-selected networks are removed.

https://github.com/brave/brave-ios/assets/5314553/ba3d38ee-6a20-49f5-871e-ecc75195ab86

### Test `Reset` & persistence
1. Go to Portfolio and tap the filters button.
2. Change some filters and tap `Save Changes`.
3. Close wallet and/or Brave app
4. Re-open the app to Portfolio,  and tap the filters button.
5. Verify the filters were preserved
6. Tap `Reset`
7. Verify the filters were reset. Defaults:
    - `Sort Assets`: High to Low
    - `Hide Small Balances`: Off/disabled
    - `Select Accounts`: all accounts selected
    - `Select Networks`: all networks except test networks (when enabled) are selected

https://github.com/brave/brave-ios/assets/5314553/4e31ec0f-f822-4129-8192-d7bb04aca086

## Screenshots:

Light Mode | Dark Mode
--|--
![Filters   Display Settings (light)](https://github.com/brave/brave-ios/assets/5314553/0efb85c1-17e7-4968-8172-cfe9688d6b0d) | ![Filters   Display Settings (dark)](https://github.com/brave/brave-ios/assets/5314553/81d91a32-c4b8-46dd-8e8b-a917637487da)
![Account Filter (light)](https://github.com/brave/brave-ios/assets/5314553/21b7c3cf-b8dc-4ba1-917c-739d35e852a3) | ![Account Filter (dark)](https://github.com/brave/brave-ios/assets/5314553/5d0d3a86-1534-49e4-8971-c46561e03111)
![Network Filter (light)](https://github.com/brave/brave-ios/assets/5314553/eba1b157-8dbb-4805-b0f9-a6d66da415ab) | ![Network Filter (dark)](https://github.com/brave/brave-ios/assets/5314553/8c6434ac-d81f-4244-bf84-0623e00d0155)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
